gtest-parallel

```diff
@@ -82,6 +87,45 @@ class RawFormat:
   def end(self):
     pass
 
+# Record of test runtimes. Has built-in locking.
+class TestTimes(object):
+  def __init__(self, save_file):
+    "Create new object seeded with saved test times from the given file."
+    self.__old_times = {}  # (test binary, test name) -> runtime in seconds
+    self.__new_times = {}  # (test binary, test name) -> runtime in seconds
+
+    # Protects __new_times; __old_times doesn't need locking since
+    # it's read-only after construction.
+    self.__lock = threading.Lock()
+
+    try:
+      with gzip.GzipFile(save_file, "rb") as f:
+        self.__old_times = cPickle.load(f)
```

Comment:
Is there any chance that you can validate that old_times is now a (test binary, test name) -> runtime in seconds? Or at least (string, string) -> int/float?